src/libcore/array.rs

```diff
@@ -406,6 +408,73 @@ mod impls_using_const_generics {
         }
     }
 
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    impl<T, const N: usize> IntoIterator for [T; N]
+    where
+        [T; N]: LengthAtMost32,
+    {
+        type Item = T;
+        type IntoIter = IntoIter<T, {N}>;
+
+        fn into_iter(self) -> Self::IntoIter {
+            IntoIter {
+                data: ManuallyDrop::new(self),
+                pos: 0,
+            }
+        }
+    }
+
+    /// A by-value array iterator.
+    ///
+    /// This struct is created by the `IntoIterator::into_iter`] method on
+    /// arrays.
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    pub struct IntoIter<T, const N: usize> {
+        /// This is the array we are iterating over. Elements that were already
+        /// yielded (the ones with index `< self.pos`) have been read via
```

Comment:
```suggestion
        /// yielded (the ones with `index < self.pos`) have been read via
```
(Looks a bit more natural I think)